pkg/controllers/managementuser/nodesyncer/nodessyncer.go

```diff
@@ -879,6 +879,13 @@ func (m *nodesSyncer) convertNodeToMachine(node *corev1.Node, existing *apimgmtv
 	cleanStatus(machine)
 	apimgmtv3.NodeConditionRegistered.True(machine)
 	apimgmtv3.NodeConditionRegistered.Message(machine, "registered with kubernetes")
+
+	// remove the "Drained" condition from the machine's status conditions
+	// only if the machine is schedulable (i.e., not unschedulable).
+	if !machine.Spec.InternalNodeSpec.Unschedulable {
+		removeDrainCondition(machine)
```

Comment:
We also need to reset `machine.Spec.DesiredNodeUnschedulable`. More context here: https://github.com/rancher/rancher/pull/48739#discussion_r1919563366

Reply:
I updated the code as per your suggestion.